processing/src/main/java/io/druid/segment/data/ListIndexed.java

```diff
@@ -67,4 +69,10 @@ public int indexOf(T value)
   {
     return baseList.iterator();
   }
+
+  @Override
+  public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+  {
+    inspector.visit("baseList", baseList);
```

Comment:
It would make it too easy to forget to annotate some fields. For the same reason `inspectRuntimeShape()` method doesn't have default Java 8 implementation with empty body, although it is always correct and a lot of implementations just do that: to force developers to implement and think about the runtime shape.

It's just my opinion, though. Maybe it makes sense to add annotation-based runtime shape inspection in some form. But anyway, not as part of this PR.